analyzers/src/SonarAnalyzer.Common/Helpers/KnownMethods.cs

```diff
@@ -31,194 +31,162 @@ public static bool IsMainMethod(this IMethodSymbol methodSymbol)
+            methodSymbol != null
+            && (methodSymbol.IsOverride || methodSymbol.IsInType(KnownType.System_Object))
+            && methodSymbol.MethodKind == MethodKind.Ordinary
+            && methodSymbol.Name == nameof(object.ToString)
+            && methodSymbol.Parameters.Length == 0
+            && methodSymbol.ReturnType.Is(KnownType.System_String);
 
         public static bool IsIDisposableDispose(this IMethodSymbol methodSymbol)
         {
             const string explicitName = "System.IDisposable.Dispose";
-            return methodSymbol != null &&
-                methodSymbol.ReturnsVoid &&
-                methodSymbol.Parameters.Length == 0 &&
-                (methodSymbol.Name == nameof(IDisposable.Dispose) ||
-                 methodSymbol.Name == explicitName);
+            return methodSymbol != null
+                && methodSymbol.ReturnsVoid
+                && methodSymbol.Parameters.Length == 0
+                && (methodSymbol.Name == nameof(IDisposable.Dispose) || methodSymbol.Name == explicitName);
         }
 
         public static bool IsIEquatableEquals(this IMethodSymbol methodSymbol)
         {
             const string explicitName = "System.IEquatable.Equals";
-            return methodSymbol != null &&
-                methodSymbol.Parameters.Length == 1 &&
-                methodSymbol.ReturnType.Is(KnownType.System_Boolean) &&
-                (methodSymbol.Name == nameof(object.Equals) ||
-                methodSymbol.Name == explicitName);
+            return methodSymbol != null
+                && methodSymbol.Parameters.Length == 1
+                && methodSymbol.ReturnType.Is(KnownType.System_Boolean)
+                && (methodSymbol.Name == nameof(object.Equals) || methodSymbol.Name == explicitName);
         }
 
         public static bool IsGetObjectData(this IMethodSymbol methodSymbol)
         {
             const string explicitName = "System.Runtime.Serialization.ISerializable.GetObjectData";
-            return methodSymbol != null &&
-                methodSymbol.Parameters.Length == 2 &&
-                methodSymbol.Parameters[0].Type.Is(KnownType.System_Runtime_Serialization_SerializationInfo) &&
-                methodSymbol.Parameters[1].Type.Is(KnownType.System_Runtime_Serialization_StreamingContext) &&
-                methodSymbol.ReturnsVoid &&
-                (methodSymbol.Name == "GetObjectData" ||
-                methodSymbol.Name == explicitName);
+            return methodSymbol != null
+                && methodSymbol.Parameters.Length == 2
+                && methodSymbol.Parameters[0].Type.Is(KnownType.System_Runtime_Serialization_SerializationInfo)
+                && methodSymbol.Parameters[1].Type.Is(KnownType.System_Runtime_Serialization_StreamingContext)
+                && methodSymbol.ReturnsVoid
+                && (methodSymbol.Name == "GetObjectData" || methodSymbol.Name == explicitName);
```

Comment:
I think you can move name checks before the Parameters Type checks, right? 

and maybe the same for all name checks which are done after `Type.Is()`  (e.g. lines 109,178,182) - string comparisons should be cheaper than the type checks which eventually [do multiple comparisons](https://github.com/SonarSource/sonar-dotnet/blob/6ef1c842c09232c065b477f79910effb8554523f/analyzers/src/SonarAnalyzer.Common/Helpers/TypeHelper.cs#L60)